pkg/securitycontextconstraints/sccmatching/provider_test.go

```diff
@@ -1004,6 +1007,66 @@ func TestValidateAllowedVolumes(t *testing.T) {
 	}
 }
 
+func TestValidateProjectedVolume(t *testing.T) {
+	pod := defaultPod()
+	scc := defaultSCC()
+	provider, err := NewSimpleProvider(scc)
+	require.NoError(t, err, "error creating provider")
+
+	tests := []struct {
+		desc                  string
+		allowedFSTypes        []securityv1.FSType
+		projectedVolumeSource *api.ProjectedVolumeSource
+		wantAllow             bool
+	}{
+		{
+			desc:                  "deny if secret is not allowed",
+			allowedFSTypes:        []securityv1.FSType{securityv1.FSTypeEmptyDir},
+			projectedVolumeSource: serviceAccountTokenVolumeSource(),
+			wantAllow:             false,
+		},
+		{
+			desc:           "deny if the projected volume has volume source other than the ones in projected volume injected by service account token admission plugin",
+			allowedFSTypes: []securityv1.FSType{securityv1.FSTypeSecret},
+			projectedVolumeSource: &api.ProjectedVolumeSource{
+				Sources: []api.VolumeProjection{
+					{
+						ConfigMap: &api.ConfigMapProjection{
```

Comment:
would like to see any variation of the serviceAccountTokenVolumeSource, i.e. combinations where most is like in a service account, but one detail is changed in each test case.

Reply:
added a bunch of unit tests to the set